### DISCUSSION TITLE
Different behavior for highlights coming from external pdf program?  And Delete?

### DISCUSSION THREAD

**Discussion by notuntoward** [RyotaUshio/obsidian-pdf-plus#111] [Q&A]
I very much like that it's possible to make a linked note for a pdf highlight made by an external pdf program.  If I click on it, I see this menu:

![image](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/3851891/175a4e3a-7bcf-4b0a-ac0c-11d7771a318d)

Then I can copy a link pointing to the highlight by clicking the copy button and then paste it into my obsidian note.  But once that's done, clicking on this highlight behaves differently than clicking on the highlights made by PDF++:

-  Behavior: External highlights copied to obsidian
   - single click gives copy/edit/delete menu, as before.
   - double click goes to page
   - middle click: nothing
- Behavior: highlights made by ++PDF
  - single click: nothing
  - double click: go to page
  - middle click: change color menu

I'd like it if highlights behaved the same, no matter where they came from, especially since I'm looking for a way to delete a highlight, and that seems possible only if I made it using an external program.

**Comment by RyotaUshio** [comment]
Each highlight that is not written in the PDF file is associated with a link in some markdown file pointing to the text selection, and you can delete the highlight by deleting that link.

What do you expect to happen when you "delete" a PDF++-made highlight?
The associated link should be deleted, but if the link was copied as a callout, removing only the link without touching the surrounding callout will end up with a strange result. How should we handle these problems?

---

Personally, I don't fully understand why you want them to behave the same. If you want all the highlights to behave as external-app-made highlights, you can just always write highlights to files. Is there any particular reason?

**Reply by notuntoward** [reply]
I'd expect "delete" to do what it does on other pdf annotators: make the highlight disappear.  

The first time I read a paper, I usually go through it fast, and make a lot of highlights.  Then, if the paper looks useful, I go through it again, taking notes, and deleting the highlights that, with better understanding, I can see are superfluous.  I don't want to see them anymore.

What should happen to any obsidian page link that's associated with a deleted highlight?  I think I would like the link to also disappear.  This is in the opposite direction: PDF++ deletes highlights by deleting links; I'd rather delete links by deleting highlights.  This is because, on my second trip through the paper, the paper is what I'm looking at.  I don't really have any "notes" to look at yet, just links.  Maybe  deleting in both directions is the way to go?

Why do I want highlights copied from an external reader to behave like highlights made by PDF++?  For one, it's because they look exactly the same -- when they don't do the same thing when I click on them, there's a bit of mental friction. I have to say to myself "this must have come from PDF++", and then go on a hunt through the obsidian note to find the link and delete it.  It would seem quicker and less distracting to just click on the highlight and delete it there.
